### DISCUSSION TITLE
Nextjs 14 too slow on localhost [Windows 10]

### DISCUSSION THREAD

**Discussion by Shashankd48** [vercel/next.js#67198] [Show and tell]
My project is built on Nodejs 18.18.1 and npm 9.8.1
![Screenshot 2024-06-26 110016](https://github.com/vercel/next.js/assets/34672115/ace8d5ff-a51c-4cd4-895a-03fcedf4a2d6)

And it really slow when running on localhost, switching pages take 5-15 sec and its just frontend no API is being used for now its static content.

The only dymanic content is language I'm using middleware and i18next for multilingual.

**Nextjs Version 14
Settings  - App router**

![Screenshot 2024-06-26 110413](https://github.com/vercel/next.js/assets/34672115/96c8003d-13bb-4cc8-b694-3eea428a4095)

its even slower in browser I'm using Link from next/link and when i click on a link it takes 4-10 sec to compile and more time to reflect in browser

**Package.json**

{
   "name": "matchare-frontend",
   "version": "0.1.0",
   "private": true,
   "scripts": {
     "dev": "next dev",
     "build": "next build",
     "start": "next start -p 3001",
     "lint": "next lint"
   },
   "dependencies": {
     "@nextui-org/react": "^2.2.10",
     "@reduxjs/toolkit": "^2.0.1",
     "aos": "^3.0.0-beta.6",
     "axios": "^1.6.8",
     "chart.js": "^3.9.1",
     "chartjs-plugin-dragdata": "^2.2.5",
     "clsx": "^2.0.0",
     "date-fns": "^3.0.0",
     "debounce": "^2.0.0",
     "formik": "^2.4.5",
     "framer-motion": "^11.0.12",
     "js-cookie": "^3.0.5",
     "jsoneditor": "^10.0.2",
     "jwt-decode": "^4.0.0",
     "lodash.debounce": "^4.0.8",
     "next": "14.0.4",
     "qrcode": "^1.5.3",
     "react": "^18",
     "react-dom": "^18",
     "react-hot-toast": "^2.4.1",
     "react-redux": "^9.0.4",
     "redux-persist": "^6.0.0",
     "swiper": "^11.0.5",
     "tailwind-merge": "^2.1.0",
     "usehooks-ts": "^2.9.1",
     "yup": "^1.3.3"
   },
   "devDependencies": {
     "@iconify/react": "^4.1.1",
     "@types/aos": "^3.0.7",
     "@types/js-cookie": "^3.0.6",
     "@types/jsoneditor": "^9.9.5",
     "@types/lodash": "^4.17.0",
     "@types/node": "^20",
     "@types/qrcode": "^1.5.5",
     "@types/react": "^18",
     "@types/react-dom": "^18",
     "autoprefixer": "^10.0.1",
     "eslint": "^8",
     "eslint-config-next": "14.0.4",
     "postcss": "^8",
     "tailwindcss": "^3.3.0",
     "typescript": "^5"
   }
 }

**Comment by sami-ahmad-1** [comment]
I am also  facing the same problem , it takes 40-50 seconds to reload the changes

**Reply by khaukheng** [reply]
Same, it's painfully slow, did you find anything to solve the issues?

EDIT: For me, I have found my issue, it's because I am using Dynamic Icon Component from lucide-react that is causing the extreme lag in my local dev environment.

The comment here solves my issue, dropping the link here in case someone encounter the same issue with mine.
[lucide dynamic icon issue](https://github.com/lucide-icons/lucide/issues/1576#issuecomment-2335019821)

**Comment by Southclaws** [comment]
This could be related to the stock anti virus built into Windows, as it's not optimised for developer tools out of the box. I think if you exclude certain drives or directories from Defender's file scanning, it'll remove filesystem trampolines from system calls leading to faster reads/writes. From what I remember, Next.js performs a metric ton of reads and writes during builds for various things like HMR and chunk/build caching.
